File: peek_core_device/_private/tuples/GpsLocationUpdateTupleAction.ts
import { addTupleType, Tuple } from "@synerty/vortexjs"
import { deviceTuplePrefix } from "./_private/PluginNames"

export enum GpsLocationUpdateTypeEnum {
    ACCURACY_COARSE = 1,
    ACCURACY_FINE = 2
}

@addTupleType
export class GpsLocationUpdateTupleAction extends Tuple {
    public static readonly tupleName = deviceTuplePrefix + "GpsLocationUpdateTupleAction"
    
    // This field allows customer specific data, that peek doesn't need to work
    data: { [key: string]: any } = {}
    
    latitude: number
    longitude: number
    updateType: GpsLocationUpdateTypeEnum
    
    constructor() {
        super(GpsLocationUpdateTupleAction.tupleName)
    }
}

File: peek_core_device/_private/tuples/GpsLocationUpdateTupleAction.py
from typing import Optional

from vortex.Tuple import Tuple
from vortex.Tuple import TupleField
from vortex.Tuple import addTupleType
from vortex.TupleAction import TupleActionABC

from peek_core_device._private.PluginNames import deviceTuplePrefix


@addTupleType
class GpsLocationUpdateTupleAction(TupleActionABC):
    __tupleType__ = deviceTuplePrefix + "GpsLocationUpdateTupleAction"
    ACCURACY_COARSE = 1
    ACCURACY_FINE = 2

    latitude: float = TupleField()
    longitude: float = TupleField()
    updateType: int = TupleField()
